integrations/linear/src/events.ts

```diff
@@ -0,0 +1,376 @@
+  runProperties: (payload) => [{ label: "Change action", text: payload.action }],
+};
+
+export const onIssueUpdate: EventSpecification<IssueEvent> = {
+  name: "Issue",
+  title: "On Issue Update",
+  source: "linear.app",
+  icon: "linear",
+  filter: {
+    action: ["update"],
+  },
+  parsePayload: (payload) => payload as IssueEvent,
+  runProperties: (payload) => [{ label: "Change action", text: payload.action }],
+};
+
+export const onIssueLabel: EventSpecification<IssueLabelEvent> = {
+  name: "IssueLabel",
+  title: "On IssueLabel",
+  source: "linear.app",
+  icon: "linear",
+  parsePayload: (payload) => payload as IssueLabelEvent,
+  runProperties: (payload) => [{ label: "Change action", text: payload.action }],
+};
+
+export const onIssueLabelCreate: EventSpecification<IssueLabelEvent> = {
+  name: "IssueLabel",
+  title: "On IssueLabel",
+  source: "linear.app",
+  icon: "linear",
+  filter: {
+    action: ["create"],
+  },
+  parsePayload: (payload) => payload as IssueLabelEvent,
+  runProperties: (payload) => [{ label: "Change action", text: payload.action }],
+};
+
+export const onIssueLabelRemove: EventSpecification<IssueLabelEvent> = {
+  name: "IssueLabel",
+  title: "On IssueLabel Remove",
+  source: "linear.app",
+  icon: "linear",
+  filter: {
+    action: ["remove"],
+  },
+  parsePayload: (payload) => payload as IssueLabelEvent,
+  runProperties: (payload) => [{ label: "Change action", text: payload.action }],
+};
+
+export const onIssueLabelUpdate: EventSpecification<IssueLabelEvent> = {
+  name: "IssueLabel",
+  title: "On IssueLabel Update",
+  source: "linear.app",
+  icon: "linear",
+  filter: {
+    action: ["update"],
+  },
+  parsePayload: (payload) => payload as IssueLabelEvent,
+  runProperties: (payload) => [{ label: "Change action", text: payload.action }],
+};
+
+export const onProject: EventSpecification<ProjectEvent> = {
+  name: "Project",
+  title: "On Project",
+  source: "linear.app",
+  icon: "linear",
+  parsePayload: (payload) => payload as ProjectEvent,
+  runProperties: (payload) => [{ label: "Change action", text: payload.action }],
+};
+
+export const onProject_Create: EventSpecification<ProjectEvent> = {
+  name: "Project",
+  title: "On Project",
+  source: "linear.app",
+  icon: "linear",
+  filter: {
+    action: ["create"],
+  },
+  parsePayload: (payload) => payload as ProjectEvent,
+  runProperties: (payload) => [{ label: "Change action", text: payload.action }],
+};
+
+export const onProject_Remove: EventSpecification<ProjectEvent> = {
+  name: "Project",
+  title: "On Project Remove",
+  source: "linear.app",
+  icon: "linear",
+  filter: {
+    action: ["remove"],
+  },
+  parsePayload: (payload) => payload as ProjectEvent,
+  runProperties: (payload) => [{ label: "Change action", text: payload.action }],
+};
+
+// TODO: think of a better naming scheme (clashes with ProjectUpdate entity)
+export const onProject_Update: EventSpecification<ProjectEvent> = {
```

Comment:
Can anyone think of a sensible naming scheme?

`Project` and `ProjectUpdate` entities exist.

onProjectUpdate - any ProjectUpdate changes
onProjectUpdate - a Project update
onProjectUpdateUpdate - a ProjectUpdate update

:exploding_head: